src/components/editing-buttons.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import { MODES } from '../modes';
+import UserInfo from '../stores/user-info';
+import UserAction from '../actions/user-action';
+
+export default function EditingButtons (props) {
+    const { disabled, currentUser, isEditing, onChange } = props;
+
+    const _handleClickUndo = e => {
+        onChange({ mode: MODES.default, formErrors: {} });
+        e.preventDefault();
+    };
+
+    const _handleClickSave = e => {
```

Comment:
AFAIK It's uncommon way to implement handlers.
Each time on redering handlers will be created again. It means overheads.
Try to use `useCallback` hook https://ru.reactjs.org/docs/hooks-reference.html#usecallback
or rewrite this component into class